packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -513,6 +513,487 @@ export class CloudFormationDeleteStackAction extends CloudFormationDeployAction
   }
 }
 
+/**
+ * Determines how IAM roles are created and managed
+ */
+export enum StackSetPermissionModel {
+  /**
+   * You must create administrator and execution roles to deploy to target accounts
+   */
+  SELF_MANAGED = 'SELF_MANAGED',
+
+  /**
+   * AWS CloudFormation StackSets automatically creates the IAM roles required to deploy to accounts managed by AWS Organizations. This requires an
+   * account to be a member of an Organization
+   */
+  SERVICE_MANAGED = 'SERVICE_MANAGED'
+}
+
+/**
+ * Describes whether AWS CloudFormation StackSets automatically deploys to AWS Organizations accounts that are added to a target organization or
+ * organizational unit (OU)
+ */
+export enum StackSetOrganizationsAutoDeployment {
+  /**
+   * StackSets automatically deploys additional stack instances to AWS Organizations accounts that are added to a target organization or
+   * organizational unit (OU) in the specified Regions. If an account is removed from a target organization or OU, AWS CloudFormation StackSets
+   * deletes stack instances from the account in the specified Regions
+   */
+  ENABLED = 'Enabled',
+
+  /**
+   * StackSets does not automatically deploy additional stack instances to AWS Organizations accounts that are added to a target organization or
+   * organizational unit (OU) in the specified Regions
+   */
+  DISABLED = 'Disabled',
+
+  /**
+   * Stack resources are retained when an account is removed from a target organization or OU
+   */
+  ENABLED_WITH_STACK_RETENTION = 'EnabledWithStackRetention'
+}
+
+/**
+ * A StackSet parameter
+ */
+export interface StackSetParameter {
+  /**
+   * The name of the parameter
+   */
+  readonly parameterKey: string
+
+  /**
+   * The parameter value.
+   *
+   * @default No default value
+   */
+  readonly parameterValue?: string
+
+  /**
+   * Use previous value? If this is specified, parameterValue needn't be specified for a StackSet update
+   *
+   * @default false
+   */
+  readonly usePreviousValue?: boolean
+}
```

Comment:
Not a huge fan of this API. Seems extremely wordy.

How about we just make this a map of string keys to values, and if the value provided is `undefined` / `null`, we set `usePreviousValue` to `true`? Should be much more concise.